### ISSUE TITLE
Collecting/loading/processing batch from data with different resolution

### ISSUE THREAD

**Issue by Masaaki-75** [Project-MONAI/MONAI#8204]
### Is your feature request related to a problem? Please describe.
I have collected a bunch of CT volumes, each is stored in `.nii.gz` format and is of different resolutions (in depth, height and width). Due to the different resolutions, when i was trying to load them using pytorch's default DataLoader, i had to set the batch size to 1, which can be inefficient and cannot fully utilize the GPU memory. 

### Describe the solution you'd like
For efficient network training, i am expecting a certain class of Dataset/DataLoader that can load several volumes (with different resolutions) as a batch efficiently, and (randomly) crop them into the same-sized sub-volumes to feed in the network.

Are there any designs in MONAI that can address this?

**Comment by KumoLiu** [comment]
Hi @Masaaki-75, thanks for your interest here.
To load and preprocess data of any size, you can use `LoadImage`. After loading, before batching, you can apply `ResizeWithPadOrCrop` to make sure all images have the same dimensions.
Alternatively, you can use `pad_list_data_collate` in `DataLoader` to automatically pad each item to match the shape of the largest tensor in each dimension. Here’s an example of how to use it:
https://github.com/Project-MONAI/tutorials/blob/2db8c620ae7e9e8f28cb2c7abb9e6cc75e3bc5c1/modules/interpretability/cats_and_dogs.ipynb#L98

Hope it helps, thanks.

**Comment by Masaaki-75** [comment]
I see. Just tried it on my training pipeline and it worked! Better yet, it seems to be compatible with other popular pytorch libraries like `accelerate`. Thank you!

**Comment by Masaaki-75** [comment]
Hi @KumoLiu , i am considering an advanced feature for data loading and processing: resizing (by cropping/padding/interpolation/..., whatever) the batch to different resolutions in each iteration. 

For example, given two preset resolutions like `[128, 256, 256]` and `[256, 512, 512]`, each batch will be resized to one of these resolutions randomly. The main idea is for data augmentation and for forcing the model to adapt to inputs with varying resolution.

Are there any similar cases in MONAI tutorials or existing solutions to this problem? Thanks!

**Comment by Masaaki-75** [comment]
Update: one possible solution is the `RandSpatialCropd`, which enables arbitrary random shapes sampled between `roi_size` and `max_roi_size`. But it seems that this class does not support using only a few preset shapes, so my current solution is simply applying a `RandSpatialCropd` with a large `roi_size` (`max_roi_size` not assigned) in the dataset, and then run `F.interpolate` after the batch is loaded during the training loop.
